deltacat/compute/compactor/steps/materialize.py

```diff
@@ -154,8 +149,7 @@ def _materialize(
             src_file_idx_np.item(),
             file_reader_kwargs_provider=read_kwargs_provider,
         )
-        record_count = len(pa_table)
```

Comment:
This was a bug initially which counted record counts of original `pa_table` and not the filtered table. We may need to setup unit test and integration to catch these kind of issues going forward.